matrix-mult.futil

```diff
@@ -0,0 +1,234 @@
+import "primitives/core.futil";
+import "primitives/memories/comb.futil";
+import "primitives/binary_operators.futil";
+
+// primitive comb_mem_d2[WIDTH, D0_SIZE, D1_SIZE, D0_IDX_SIZE, D1_IDX_SIZE]
+component main() -> () {
+  cells {
+    @external(1) mem1 = comb_mem_d2(32, 2, 2, 1, 1); //matrix 1, and will store the result
+    @external(1) mem2 = comb_mem_d2(32, 2, 2, 1, 1); //matrix 2
+    val = std_reg(32); // holds one number of the result matrix
+    val1 = std_reg(32);
+    val2 = std_reg(32);
+    val3 = std_reg(32);
+    val4 = std_reg(32);
+    val5 = std_reg(32);
+    val6 = std_reg(32);
+    val7 = std_reg(32);
+    val8 = std_reg(32);
+    temp1 = std_reg(32);
+    temp2 = std_reg(32);
+
+    add = std_add(32);
+    mul = std_mult_pipe(32);
+  }
+
+  wires {
+    group read1 { // read value 00 from matrix 1
```

Comment:
Sometime I find it useful to have pseudocode in my comments. So here you could just have `val1 := mem1[0][0]`